store/interface.go

```diff
@@ -99,6 +99,7 @@ type Reader interface {
 	LastCertificate() *certificate.BlockCertificate
 	IsBanned(addr crypto.Address) bool
 	Prune(resultFunc func(pruned, skipped, pruningHeight uint32)) error
+	PruneOnCommit() error
```

Comment:
We don't need to tell the store to prune the block; the store should know when to prune or not. Let each module do its job